src/addDebouncedCopy.tsx

```diff
@@ -0,0 +1,62 @@
+import {
+  flowMax,
+  addState,
+  addHandlers,
+  addEffect,
+  CurriedPropsAdder,
+} from 'ad-hok'
+
+import addPropTrackingRef from './addPropTrackingRef'
+import addDebouncedHandler from './addDebouncedHandler'
+import cleanupProps from './cleanupProps'
+
+type AddDebouncedCopyType = <
+  TProps extends {},
+  TPropName extends keyof TProps,
+  TDebouncedPropName extends string
+>(
+  waitInterval: number,
+  propName: TPropName,
+  debouncedPropName: TDebouncedPropName,
+) => CurriedPropsAdder<
+  TProps,
+  {
+    [PropName in TDebouncedPropName]: TProps[TPropName]
+  }
+>
+
+const addDebouncedCopy: AddDebouncedCopyType = (
+  waitInterval,
+  propName,
+  debouncedPropName,
+) => {
+  const setterName = 'addDebouncedCopySetter'
+  const refName = 'addDebouncedCopyRef'
+
+  return flowMax(
+    addState(
+      debouncedPropName,
+      setterName,
+      ({[propName]: propValue}) => propValue,
+    ),
+    addPropTrackingRef(propName, refName),
+    addHandlers(
+      {
+        [setterName]: ({[refName]: ref, [setterName]: setter}) => () => {
+          setter(ref.current)
+        },
+      },
+      [],
+    ),
+    addDebouncedHandler(waitInterval, setterName),
+    addEffect(
+      ({[setterName]: debouncedSetter}) => () => {
+        debouncedSetter()
+      },
+      [propName],
+    ),
+    cleanupProps([setterName, refName]),
```

Comment:
I added `cleanupProps()` per our discussion and tried to use consistently in these helpers (seems nice for helpers from a library not to expose stray unexpected props)

Re the "danger" of it, I think between documenting its expected/correct usage in this library's README + adding a linter rule per Peter's suggestion that'll be fine and doesn't need a `dangerous*` name